memiavl/db.go

```diff
@@ -104,10 +112,10 @@ func (db *DB) ApplyUpgrades(upgrades []*TreeNameUpgrade) error {
 	return nil
 }
 
-// Commit wraps `MultiTree.ApplyChangeSet` to add some db level operations:
-// - manage background snapshot rewriting
-// - write WAL
-func (db *DB) Commit(changeSets []*NamedChangeSet) ([]byte, int64, error) {
+// cleanupSnapshotRewrite cleans up the old snapshots and switches to a new multitree
+// if a snapshot rewrite is in progress. It returns true if a snapshot rewrite has completed
+// and false otherwise, along with any error encountered during the cleanup process.
+func (db *DB) cleanupSnapshotRewrite() (bool, error) {
 	if db.snapshotRewriteChan != nil {
```

Comment:
It looks better to quit early when `if db.snapshotRewriteChan == nil {`, instead of a big block of if clause.